### ISSUE TITLE
Upgraded Sentiment Analysis - question about sigmoid 

### ISSUE THREAD

**Issue by TalitaAnthonio** [bentrevett/pytorch-sentiment-analysis#75]
To get the predictions you do this: 

```
rounded_preds = torch.round(torch.sigmoid(preds))
```

I was wondering why I get very different accuracies when I add a sigmoid layer to the LSTM instance: 

```
self.sigmoid = nn.Sigmoid()

```

and then return the following: 

```
self.sigmoid(self.fc(hidden))
```

where self.fc is still the Linear hidden layer that you are using (and I do not apply sigmoid then in) 
```
rounded_preds = torch.round(preds)
````

I expected that the results would be the same because I am doing the same, but apparently it's something totally different. Can you explain why this is the case? 

And does it also mean that if you want to add another layer to the classifier (say for instance a RELU layer) you have to do it before feeding the results to the linear layer (so between LSTM and linear layer).

**Comment by TalitaAnthonio** [comment]
Note: I realise though that if you have a sigmoid value as the returned prediction value by the classifier instance itself, then this value will also be used to compute the loss and perform propagation, whereas if there is just a linear layer the value used to compute the loss will be the one provided by the linear layer. So it makes sense that the outcomes are different. 

So I guess that my question is then why can't we not let the classifier object return a value provided by the sigmoid function/layer (instead of using a value provided from a linear layer) or feed the output from the linear layer to a sigmoid layer and return that as the final prediction of the classifier object.

**Comment by TalitaAnthonio** [comment]
I think that I know the answer: 

[(https://discuss.pytorch.org/t/bceloss-vs-bcewithlogitsloss/33586/4)]

You are using BCEWithLogitsLoss, so the sigmoid activation function will be applied internally when the loss is computed. Therefore, you should not feed probabilities into the loss function (in contrast to BCEloss). 

However, to interpret the results / make a final prediction, we still need to convert the predicted values by the linear layer to sigmoids, which is why you do that in the ```binary_accuracy``` function.

**Comment by bentrevett** [comment]
Yep, that's exactly correct.
